File: backend/src/main/java/com/culinaryheaven/domain/recipe/dto/response/RecipeLikeResponse.java
package com.culinaryheaven.domain.recipe.dto.response;

import com.culinaryheaven.domain.recipe.domain.RecipeLike;
import com.culinaryheaven.domain.user.dto.response.UserInfoResponse;
import io.swagger.v3.oas.annotations.media.Schema;

public record RecipeLikeResponse(

        @Schema(description = "레시피 좋아요 ID")
        Long id,

        @Schema(description = "레시피 정보")
        RecipePreviewResponse recipe,

        @Schema(description = "유저 정보")
        UserInfoResponse user

) {

    public static RecipeLikeResponse of(RecipeLike recipeLike) {
        return new RecipeLikeResponse(recipeLike.getId(), RecipePreviewResponse.of(recipeLike.getRecipe()), UserInfoResponse.of(recipeLike.getUser()));
    }

}

File: android/data/src/main/java/com/jae464/data/remote/model/response/recipe/RecipeLikeResponse.kt
package com.jae464.data.remote.model.response.recipe

import com.jae464.data.remote.model.response.user.UserInfoResponse
import kotlinx.serialization.Serializable

@Serializable
data class RecipeLikeResponse(
    val id: Long,
    val recipe: RecipePreviewResponse,
    val user: UserInfoResponse,
)

